### PR TITLE
Feat/Filter bar component and refactors

### PR DESCRIPTION
**Pull request by fernandozanutto**
## Description

- Adds Filter Bar component
- Refactors and prepares the component to accept the Filter Chip, which has a dropdown menu

Fixes # (issue)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) document
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have made corresponding changes to the documentation (if appropriate)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] All new and existing tests passed

### CODE REVIEW

File: ocean-components/src/main/java/br/com/useblu/oceands/adapter/OceanChipListAdapter.kt [hunk 1]
```diff
@@ -1,140 +1,120 @@
...
+            )
+
+            OceanFilterChipViewHolder(itemBinding)
+        }
+    }
 
-        return OceanChipListViewHolder(itemBinding)
+    override fun getItemViewType(position: Int): Int {
+        return when (items[position]) {
+            is OceanBasicChip -> 1
```
Comment (nordael) [inline ocean-components/src/main/java/br/com/useblu/oceands/adapter/OceanChipListAdapter.kt:50]: Será que não é melhor definir um enum aqui? Acho que deixa mais semântico o código.

File: ocean-components/src/main/res/layout/ocean_filter_bar.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+        <import type="br.com.useblu.oceands.model.OceanChip" />
+
+        <variable
+            name="chips"
+            type="List&lt;OceanChip>" />
+
+    </data>
+
+    <!-- the outer LinearLayout is needed so when including this layout it is possible to add margins without overriding the 64dp height  -->
```
Comment (nordael) [inline ocean-components/src/main/res/layout/ocean_filter_bar.xml:18]: Esse comentário é necessário?

Comment (fernandozanutto) [reply ocean-components/src/main/res/layout/ocean_filter_bar.xml:18]: coloquei o comentário pra nao acabar de alguem achar desnecessário ter um linear layout por fora e excluir futuramente

File: ocean-components/src/main/java/br/com/useblu/oceands/adapter/OceanChipListAdapter.kt (hunk 1 above)
Comment (fernandozanutto) [reply ocean-components/src/main/java/br/com/useblu/oceands/adapter/OceanChipListAdapter.kt]: fiz a alteração no ultimo commit

**Pull request merged**